Whoaverse/Whoaverse/Controllers/HomeController.cs

```diff
@@ -398,6 +398,59 @@ public ActionResult UserProfile(string id, int? page, string whattodisplay)
                 return View("UserSubmitted", paginatedUserSubmissions);
             }
 
+            // show saved                        
+            if (whattodisplay != null && whattodisplay == "saved" && User.Identity.IsAuthenticated && User.Identity.Name == id)
+            {
+                var savedSubmissions = _db.Savingtrackers.Where(u => u.UserName == id).ToDictionary(saveSubTracker => saveSubTracker.MessageId, saveSubTracker => saveSubTracker.Timestamp);
+                var savedComments = _db.Commentsavingtrackers.Where(u => u.UserName == id).ToDictionary(saveCommentTracker => saveCommentTracker.CommentId, saveCommentTracker => saveCommentTracker.Timestamp);
+
+                var submissionIds = new List<int>(savedSubmissions.Keys);
+                var commentIds = new List<int>(savedComments.Keys);
+
+                var submissionsSavedByUser = _db.Messages.Where(submission => submissionIds.Contains(submission.Id)).ToList();
+
+                var commentsSavedByUser = _db.Comments.Where(comment => commentIds.Contains(comment.Id)).ToList();
+
+                //Merge submissions and comments into one list sorted by date
```

Comment:
I don't think that linq can do datetime comparisons or merge two different lists, which is why I did it in memory. I'm new to C# so I don't have much experience with LINQ, but when I tried orderby I got errors saying it couldn't be converted to SQL.